webapi/src/main/scala/org/knora/webapi/util/standoff/StandoffTagUtilV2.scala

```diff
@@ -731,25 +735,25 @@ object StandoffTagUtilV2 {
                                 // it refers to a knora resource
                                 StandoffTagIriAttributeV2(standoffPropertyIri = propIri, value = value)
                             }
-                        } else if (propPredicates.get(OntologyConstants.KnoraBase.ObjectDatatypeConstraint).isDefined) {
+                        } else if (propPredicates.get(OntologyConstants.KnoraBase.ObjectDatatypeConstraint.toSmartIri).isDefined) {
 
                             // it is a data type property (literal)
-                            val propDataType = propPredicates(OntologyConstants.KnoraBase.ObjectDatatypeConstraint)
+                            val propDataType: PredicateInfoV2 = propPredicates(OntologyConstants.KnoraBase.ObjectDatatypeConstraint.toSmartIri)
 
                             propDataType.objects.headOption match {
-                                case Some(OntologyConstants.Xsd.String) =>
+                                case Some(SmartIriLiteralV2(SmartIri(OntologyConstants.Xsd.String))) =>
```

Comment:
I don't understand the thing about stable identifiers, but the way you're doing it is correct. I remember I had to write the method `SmartIri.unapply` to make this work.

Reply:
There's even a test for it:

https://github.com/dhlab-basel/Knora/blob/develop/webapi/src/test/scala/org/knora/webapi/util/StringFormatterSpec.scala#L685

Reply:
I used this myself:

```scala
def f(x: Int, y: Int) = x match {
  case `y` => ...
}
```
(https://www.scala-lang.org/files/archive/spec/2.11/08-pattern-matching.html#stable-identifier-patterns)

`y` holds a value that you want to match for (e.g., 2). But I do not understand how this is related to our case. Maybe because there is the return value of a function (`toSmartIri`) involved that could behave differently although the parameter stays the same?

Reply:
I have no idea. I was just happy that I could make it possible to use `SmartIri` in match-case expressions.

Reply:
`toSmartIri` involves some compiler magic, which may interfere with using it in this context:

http://docs.scala-lang.org/overviews/core/value-classes.html

But anyway, I think `case Some(SmartIriLiteralV2(SmartIri(OntologyConstants.Xsd.String)))` looks just fine. :)

Reply:
> But anyway, I think case Some(SmartIriLiteralV2(SmartIri(OntologyConstants.Xsd.String))) looks just fine. :)

Absolutely! I was *just* curious :-) 

I once had a weird issue involving match case where I got a match error at runtime although the compiler could have figured out at compile time that there was not a single possible match. Still it compiled. This is why I want to keep an eye on details ;-) But you are right, if it works, there is no use for further investigations.